setup.py

```diff
@@ -61,14 +61,13 @@
     keywords="task-scheduling parallel numpy pandas pydata",
     classifiers=[
         "Programming Language :: Python :: 3",
-        "Programming Language :: Python :: 3.6",
         "Programming Language :: Python :: 3.7",
         "Programming Language :: Python :: 3.8",
         "Programming Language :: Python :: 3.9",
     ],
     packages=packages + tests,
     long_description=open("README.rst").read() if exists("README.rst") else "",
-    python_requires=">=3.6",
+    python_requires=">=3.7",
```

Comment:
This may cause an issue in the distributed environment-3.7.yml, see
https://github.com/dask/distributed/blob/5ea74cd7ec3cf5b90c5e6d5a31fb7dc3888ea242/continuous_integration/environment-3.6.yaml#L43-L46
Needs testing

Reply:
I'm [temporarily pointing the distributed CI to this PR](https://github.com/dask/distributed/pull/4390/commits/05452f17567d74f37ab1fa0df31821332cfeccd8) over in https://github.com/dask/distributed/pull/4390. My aim is to avoid the issue you're pointing out. Let me know if this won't properly check that